enhancements/authentication/pod-security-admission-enforcement.md

```diff
@@ -0,0 +1,315 @@
+
+If a user encounters `status.violatingNamespaces` it is expected to:
+
+- resolve the violating Namespaces in order to be able to `Upgrade` or
+- set the `spec.enforcementMode=Privileged` and solve the violating Namespaces later.
+
+If a user manages several clusters and there are well known violating Namespaces, the `spec.enforcementMode=Privileged` can be set as a precaution.
+
+### Implementation Details
+
+- The `PodSecurityReadinessController` in the `cluster-kube-apiserver-operator` will manage the new API.
+- The [`Config Observer Controller`](https://github.com/openshift/cluster-kube-apiserver-operator/blob/218530fdea4e89b93bc6e136d8b5d8c3beacdd51/pkg/operator/configobservation/configobservercontroller/observe_config_controller.go#L131) must be updated to watch for the new API's status alongside the `FeatureGate`.
+- The [`PodSecurityAdmissionLabelSynchronizationController`](https://github.com/openshift/cluster-policy-controller/blob/master/pkg/cmd/controller/psalabelsyncer.go#L17-L50) must be updated to watch for the new API's status alongside the `FeatureGate`.
+- If the `FeatureGate` `OpenShiftPodSecurityAdmission` is removed from the current `FeatureSet`, the cluster must revert to its previous state.
+  It serves as a break-glass mechanism.
+
+#### PodSecurityReadinessController
+
+The `PodSecurityReadinessController` will manage the `PSAEnforcementConfig` API.
+It already collects most of the necessary data to determine whether a Namespace would fail enforcement or not to create a [`ClusterFleetEvaluation`](https://github.com/openshift/enhancements/blob/master/dev-guide/cluster-fleet-evaluation.md).
+With the `security.openshift.io/MinimallySufficientPodSecurityStandard`, it will be able to evaluate all Namespaces for failing workloads, if any enforcement would happen.
+With the `security.openshift.io/ValidatedSCCSubjectType`, it can categorize violations more accurately and create a more accurate `ClusterFleetEvaluation`.
+
+#### PodSecurity Configuration
+
+A Config Observer in the `cluster-kube-apiserver-operator` manages the Global Config for the kube-apiserver, adjusting behavior based on the `OpenShiftPodSecurityAdmission` `FeatureGate`.
+It must watch both the `status.enforcementMode` for `Restricted` and the `FeatureGate` `OpenShiftPodSecurityAdmission` to be enabled to make decisions.
+
+#### PodSecurityAdmissionLabelSynchronizationController
+
+The [PodSecurityAdmissionLabelSynchronizationController (PSA label syncer)](https://github.com/openshift/cluster-policy-controller/blob/master/pkg/psalabelsyncer/podsecurity_label_sync_controller.go) must watch the `status.enforcementMode` and the `OpenShiftPodSecurityAdmission` `FeatureGate`.
+If `spec.enforcementMode` is `Restricted` and the `FeatureGate` `OpenShiftPodSecurityAdmission` is enabled, the syncer will set the `pod-security.kubernetes.io/enforce` label.
```

Comment:
This label goes on each namespace? Or?

Reply:
On each namespace, except it is:

- a run-level 0 namespace (default, kube-system,...), an
- `openshift` pre-fixed namespace or 
- a namespace which disabled the PSA label syncer.

Reply:
I will list those Namespaces.

Reply:
Clarifying this behaviour briefly in the EP would be useful for readers IMO